sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -102,7 +102,7 @@ public Context start(String spanName, StartSpanOptions options, Context context)
             return startSuppressedSpan(context);
         }
         context = unsuppress(context);
-        if (spanKind == SpanKind.INTERNAL && !context.getData(CLIENT_METHOD_CALL_FLAG).isPresent()) {
+        if (isInternalOrClientSpan(spanKind) && !context.getData(CLIENT_METHOD_CALL_FLAG).isPresent()) {
```

Comment:
needed for suppression when we have 
- Client (public API) -> Client (public API) -> Client (HTTP) 
- Client (public API) -> Internal (public API) -> Client (HTTP) 

spans